src/boot_O2/boot_80086760.c

```diff
@@ -1,27 +1,211 @@
+/**
+ * MathF library
+ * Contains tangent function, wrappers for a number of the handwritten functions in fp, and a suite of arctangents
+ */
 #include "global.h"
+#include "fp.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/boot_80086760/func_80086760.s")
+extern f32 func_80086C70(f32 x);
+extern f32 func_80086CA8(f32 x);
+extern f32 func_80086D50(f32 x);
+extern f32 func_80086CE0(f32 x);
+extern f32 func_80086D18(f32 x);
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/boot_80086760/func_80086794.s")
+s32 gUseAtanContFrac;
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/boot_80086760/func_800867B4.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/boot_80086760/func_800867D4.s")
+/**
+ * Tangent function computed using libultra __sinf and __cosf
+ */
+// Math_FTanF
+f32 func_80086760(f32 x) {
+    return __sinf(x) / __cosf(x);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/boot_80086760/func_800867F4.s")
+// Unused
+// Math_FFloorF
+f32 func_80086794(f32 x) {
+    func_80086C70(x);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/boot_80086760/func_80086814.s")
+// Unused
+// Math_FCeilF
+f32 func_800867B4(f32 x) {
+    func_80086CA8(x);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/boot_80086760/func_80086834.s")
+// Unused
+// Math_FRoundF
+f32 func_800867D4(f32 x) {
+    func_80086D50(x);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/boot_80086760/func_80086880.s")
+// Unused
+// Math_FTruncF
+f32 func_800867F4(f32 x) {
+    func_80086CE0(x);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/boot_80086760/func_800869A4.s")
+// Math_FNearbyIntF
+f32 func_80086814(f32 x) {
+    func_80086D18(x);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/boot_80086760/func_80086AF0.s")
+/**
+ * Arctangent approximation using a Maclaurin series [https://mathworld.wolfram.com/MaclaurinSeries.html]
+ * (one quadrant, i.e. |x| < 1)
+ */
+// Math_FAtanTaylorQF
+f32 func_80086834(f32 x) {
+    // Coefficients of Maclaurin series of arctangent
+    static const f32 coeffs[] = {
+        -1.0f / 3, +1.0f / 5, -1.0f / 7, +1.0f / 9, -1.0f / 11, +1.0f / 13, -1.0f / 15, +1.0f / 17, 0.0f,
+    };
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/boot_80086760/func_80086B30.s")
+    f32 poly = x;
+    f32 sq = SQ(x);
+    f32 exp = x * sq;
+    const f32* c = coeffs;
+    f32 term;
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/boot_80086760/func_80086C18.s")
+    // Calculate the series until adding more terms does not change the float
+    while (1) {
+        term = *c++ * exp;
```

Comment:
I thought we were programming in `c`, not `c++` (no actual suggestion, just a funny line)